### PR TITLE
Add test case for RavenDB's AV issue

### PR DESCRIPTION
**Pull request by EgorBo**
Addresses https://github.com/dotnet/runtime/pull/87847#discussion_r1272700793

David's minimal repro for the issue.

### CODE REVIEW

File: src/tests/JIT/Regression/JitBlue/Runtime_87597/Runtime_87597.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <Optimize>True</Optimize>
+  </PropertyGroup>
+  <ItemGroup>
+    <Compile Include="$(MSBuildProjectName).cs" />
+    <CLRTestEnvironmentVariable Include="DOTNET_TieredCompilation" Value="1" />
+    <CLRTestEnvironmentVariable Include="DOTNET_TieredPGO" Value="1" />
```
Comment (jakobbotsch) [inline src/tests/JIT/Regression/JitBlue/Runtime_87597/Runtime_87597.csproj:8]: I think you should remove these, or you should specify that the test requires process isolation as well.

Comment (EgorBo) [reply src/tests/JIT/Regression/JitBlue/Runtime_87597/Runtime_87597.csproj]: Thanks, will remove and run pgo pipelines (TC and PGO are enabled there)

**Review by jakobbotsch — APPROVED** [review]

**Pull request merged**